src/Microsoft.Diagnostics.Runtime/src/Desktop/HandleRoot.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Microsoft.Diagnostics.Runtime.Desktop
 {
-    internal class HandleRoot : ClrRoot
+    public class HandleRoot : ClrRoot
```

Comment:
Why is this class marked public?  HandleRoot only makes it so the space complexity of enumerating roots is lower (letting us generate objects that only contain unique data for the particular root kind).

Reply:
The reason was that I wanted to project a sequence of `ClrHandle`s as `ClrRoot`s, to merge with another sequence of `ClrRoot`s.

Looking at this again I'm wondering why I didn't just use `ClrHeap.EnumerateRoots`. It seems I was influenced by the implementation of `GCRoot.EnumerateGCRoots`. The latter includes strong handles and stack roots, while the former also includes the finalizer queue and (optionally) static fields and thread locals.

I believe `ClrHeap.EnumerateRoots` will be a better fit for me. I'll close this PR.